src/safeposix/syscalls/fs_calls.rs

```
@@ -1286,14 +1379,17 @@ impl Cage {
                         "we don't support fstat on sockets yet",
```

can we fail faster here

---

not sure how to do that since we only get to know this once we read the fd enum. Would you have any ways to do so?

---

i meant the order of match block, so that error matches go first.

---

got it